python/ray/dashboard/modules/job/job_head.py

```diff
@@ -166,7 +172,8 @@ def __init__(self, config: dashboard_utils.DashboardHeadModuleConfig):
         # `JobHead` has ever used, and will not be deleted
         # from it unless `JobAgentSubmissionClient` is no
         # longer available (the corresponding agent process is dead)
-        self._agents = dict()
+        # {node_id: JobAgentSubmissionClient}
+        self._agents: Dict[NodeID, JobAgentSubmissionClient] = dict()
 
     async def get_target_agent(self) -> Optional[JobAgentSubmissionClient]:
```

Comment:
```suggestion
    async def get_target_agent(self) -> JobAgentSubmissionClient:
```

seems it can never be None?